tpu/src/main/java/tpu/CreateTimeBoundQueuedResource.java

```diff
@@ -0,0 +1,161 @@
+      String zone, String tpuType, String tpuSoftwareVersion)
+      throws IOException, ExecutionException, InterruptedException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (TpuClient tpuClient = TpuClient.create()) {
+      // Define parent for requests
+      String parent = String.format("projects/%s/locations/%s", projectId, zone);
+      // Create a Duration object representing 6 hours.
+      Duration validAfterDuration = Duration.newBuilder().setSeconds(6 * 3600).build();
+      // Uncomment the appropriate lines to use other time bound configurations
+      // Duration validUntilDuration = Duration.newBuilder().setSeconds(6 * 3600).build();
+      // String validAfterTime = "2024-10-14T09:00:00Z";
+      // String validUntilTime = "2024-12-14T09:00:00Z";
+
+      // Create a node
+      Node node =
+          Node.newBuilder()
+              .setName(nodeName)
+              .setAcceleratorType(tpuType)
+              .setRuntimeVersion(tpuSoftwareVersion)
+              .setQueuedResource(
+                  String.format(
+                      "projects/%s/locations/%s/queuedResources/%s",
+                      projectId, zone, queuedResourceName))
+              .build();
+
+      // Create queued resource
+      QueuedResource queuedResource =
+          QueuedResource.newBuilder()
+              .setName(queuedResourceName)
+              .setTpu(
+                  QueuedResource.Tpu.newBuilder()
+                      .addNodeSpec(
+                          QueuedResource.Tpu.NodeSpec.newBuilder()
+                              .setParent(parent)
+                              .setNode(node)
+                              .setNodeId(nodeName)
+                              .build())
+                      .build())
+              .setQueueingPolicy(
+                  QueuedResource.QueueingPolicy.newBuilder()
+                      // You can specify a duration after which a resource should be allocated.
+                      // corresponds   --valid-after-duration flag
+                      .setValidAfterDuration(validAfterDuration)
+                      .build())
+              // Uncomment the appropriate lines to use other time bound configurations
+              //.setQueueingPolicy(
+              //QueuedResource.QueueingPolicy.newBuilder()
+              // You can specify a time after which a resource should be allocated.
+              // corresponds  --valid-until-duration flag
+              //.setValidUntilDuration(validUntilDuration)
+              //.build())
+              //.setQueueingPolicy(
+              //QueuedResource.QueueingPolicy.newBuilder()
+              // You can specify a time before which the resource should be allocated.
+              // corresponds  --valid-after-time flag
+              //.setValidAfterTime(convertStringToTimestamp(validAfterTime))
```

Comment:
Add this line below 106, and modify it to include the validAfterTime directly.

Reply:
Thank you! Fixed